### ISSUE TITLE
Infinispan marshalling of POJOs with boolean fields not working

### ISSUE THREAD

**Issue by omasseau** [quarkusio/quarkus#42023]
### Describe the bug

Using Infinispan extension, marshalling of POJOs with simple primitive boolean fields is not working and gives this error : 

> C:\dev\git\quarkus-test\src\main\java\com\test\Person.java:22: error: org.infinispan.protostream.annotations.ProtoSchemaBuilderException: Primitive field 'isMarried' of com.test.Person is not nullable so it should be either marked required or should have a default value, while processing com.test.Schema
> interface Schema extends GeneratedSchema {
> ^
>   	at org.infinispan.protostream.annotations.impl.ProtoMessageTypeMetadata.discoverFieldsFromClassFields(ProtoMessageTypeMetadata.java:625)

Yet the infinispan documentation says : 

![image](https://github.com/user-attachments/assets/0971681a-2f46-4068-ace4-16e684986a64)

So I'm expecting the marshalling of boolean primitive type to work.

Here is the class : 

```
@Proto
public class Person {
    public String name;
    public boolean isMarried = false;

    public Person(){
    }

    public Person(String name, boolean isMarried) {
        this.name = name;
        this.isMarried = isMarried;
    }
}

@ProtoSchema(includeClasses = { Person.class })
interface Schema extends GeneratedSchema {
}
```

Is this a quarkus limitation of the Infinispan documentation is wrong ?
Having to change all my boolean fields to Boolean is not a solution. I'm expecting a framework/library to be able at least to marshall any primitive types out of the box.

### Expected behavior

POJOs with boolean fields should be able to be marshalled out of the box

### Output of `java -version`

java 21.0.3

### Quarkus version or git rev

3.12.3

**Comment by omasseau** [comment]
Note: 

It seems boolean fields work only with records.
I have no error when trying this : 

```
@Proto
public record Person (String name, boolean isMarried) {
}

@ProtoSchema(includeClasses = { Person.class })
interface Schema extends GeneratedSchema {
}
```

Still this is not a solution for me as in my case I cannot use records.

**Comment by karesti** [comment]
@omasseau have you tried adding ```java @ProtoField(defaultValue = "false")``` over the field as a workaround? 
In any case, this should work for records and classes. Opening an issue in Protostream @tristantarrant

**Comment by omasseau** [comment]
> @ProtoField(defaultValue = "false")

Hi @karesti 

Yes it works, but only if I also explicitely define a number : 
```
    @ProtoField(defaultValue = "false", number = 2)
    public boolean isMarried = false;
```

As you can guess, we don't really want to explicitely add this annotation on every boolean fields, as it somewhat defeats the expected benefits of having a protobuf schema being automatically generated for us :)

**Comment by karesti** [comment]
yes, I opened an issue @omasseau https://issues.redhat.com/browse/IPROTO-362
@fax4ever
